mmdet3d/models/necks/view_transformer.py

```diff
@@ -0,0 +1,310 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+import torch
+import torch.nn as nn
+from mmcv.runner import BaseModule
+
+from ..builder import NECKS
+
+
+class QuickCumsum(torch.autograd.Function):
+    """Back propagation accelerated cumulative sum operator."""
+
+    @staticmethod
+    def forward(ctx, x, coor, ranks):
+        x = x.cumsum(0)
+        kept = torch.ones(x.shape[0], device=x.device, dtype=torch.bool)
+        kept[:-1] = (ranks[1:] != ranks[:-1])
+
+        x, coor = x[kept], coor[kept]
+        x = torch.cat((x[:1], x[1:] - x[:-1]))
+
+        # save kept for backward
+        ctx.save_for_backward(kept)
+
+        # no gradient for coordinates
+        ctx.mark_non_differentiable(coor)
+
+        return x, coor
+
+    @staticmethod
+    def backward(ctx, gradx, gradgeom):
+        kept, = ctx.saved_tensors
+        back = torch.cumsum(kept, 0)
+        back[kept] -= 1
+
+        val = gradx[back]
+
+        return val, None, None
+
+
+@NECKS.register_module()
+class ViewTransformerLiftSplatShoot(BaseModule):
+    """Lift-Splat-Shoot view transformer for transform image-view feature into
+    bird-eye-view feature.
+
+    Args:
+        grid_config (dict(axis:list(float,float,float))): Config of grid alone
```

Comment:
No need for details of dict in the type hint
Can just explain it in the description